api/v1/services/ProjectService.js

```diff
@@ -0,0 +1,188 @@
+			if(_.isNull(result)) {
+				var message = "The project id is invalid";
+				var source = "project_id";
+				throw new errors.InvalidParameterError(message, source);
+			}
+			return Mentor.findById(mentor_id);
+		})
+		.then(function (mentor) {
+			if(_.isNull(mentor)) {
+				var message = "The mentor id is invalid";
+				var source = "mentor_id";
+				throw new errors.InvalidParameterError(message, source);
+			}
+			return _Promise.resolve(false);
+		});
+}
+
+/**
+ * Helper function for deleting project-mentor relationships
+ * @param  {Int} ID of the project assigned to the mentor
+ * @param  {Int} ID of the mentor assigned to the project
+ * @return {Promise} resolving to null
+ * @throws InvalidParameterError when a project or mentor doesn't exist with the specified ID
+ */
+_deleteProjectMentor = function (project_id, mentor_id) {
+	return ProjectMentor
+		.findByProjectAndMentorId(project_id, mentor_id)
+		.then(function(oldProjectMentor) {
+			if(_.isNull(oldProjectMentor)) {
+				var message = "A project-mentor relationship with the given IDs cannot be found";
+				var source = "project_id/mentor_id";
+				throw new errors.NotFoundError(message, source);
+			}
+			return oldProjectMentor.destroy();
+		});
+}
+
+
+/**
+ * Add a new project-mentor relationship
+ * @param  {Int} ID of the project assigned to the mentor
+ * @param  {Int} ID of the mentor assigned to the project
+ * @return {Promise} resolving to the new relationship
+ * @throws InvalidParameterError when a project or mentor doesn't exist with the specified ID
+ */
+module.exports.addProjectMentor = function (project_id, mentor_id) {
+	var projectMentor = ProjectMentor.forge({ project_id: project_id, mentor_id: mentor_id });
+
+	return _isProjectMentorValid(project_id, mentor_id)
+		.then(function (isValid) {
+			return ProjectMentor.findByProjectAndMentorId(project_id, mentor_id);
+		})
+		.then(function (result) {
+			if (!_.isNull(result)) {
+				//The project mentor relationship already exists
+				return _Promise.resolve(result);
+			}
+			return projectMentor.save()
+		});
+}
+
+/**
+ * Deletes a project-mentor relationship
+ * @param  {Int} ID of the project in question
+ * @param  {Int} ID of the mentor in question
+ * @return {Promise} resolving to the deleted relationship
+ * @throws InvalidParameterError when a project or mentor doesn't exist with the specified ID
+ */
+module.exports.deleteProjectMentor = function (project_id, mentor_id) {
+	return _deleteProjectMentor(project_id, mentor_id);
+}
+
+
+/**
+ * Returns a list of all projects
+ * @param  {Int} Page number
+ * @param  {Int} Number of items on the page
+ * @param  {Int} Boolean in int form representing published/unpublished
+ * @return {Promise} resolving to an array of project objects
+ */
+module.exports.getAllProjects = function (page, count, isPublished) {
+	return Project
+		.query(function (qb){
+			qb.groupBy('projects.id');
+			qb.where('is_published', '=', String(isPublished));
```

Comment:
Do you need to make `isPublished` a string? If it's an integer value, it should just work with `0` or `1`